test/start-kubernetes.sh

```diff
@@ -351,7 +351,27 @@ function init_kubernetes_cluster() (
     echo "Installing dependencies on cloud images, this process may take some minutes"
     vm_id=0
     pids=""
-    for ip in $(print_ips); do
+    IPS=$(print_ips)
+    # in Fedora-31 case, add 'systemd.unified_cgroup_hierarchy=0' to kernel cmdline and reboot
+    if [ "$TEST_DISTRO" = "fedora" -a "$TEST_DISTRO_VERSION" = "31" ]; then
+        for ip in ${IPS}; do
+            ssh $SSH_ARGS ${CLOUD_USER}@${ip} "sudo sed -i 's/n8\"/n8 systemd.unified_cgroup_hierarchy=0\"/g' /etc/default/grub"
+            ssh $SSH_ARGS ${CLOUD_USER}@${ip} "sudo grub2-mkconfig -o /boot/grub2/grub.cfg"
+            # use --force to speed up shutdown as nothing important running yet to wait for
+            ssh $SSH_ARGS ${CLOUD_USER}@${ip} "sudo systemctl --force reboot"
+        done
+        for ip in ${IPS}; do
+            SECONDS=0
+            echo "After reboot: Waiting for ssh connectivity on vm with ip $ip"
+            # Verify that system runs with added cmdline parameter
+            while ! ssh $SSH_ARGS ${CLOUD_USER}@${ip} "grep -q systemd.unified_cgroup_hierarchy /proc/cmdline"; do
+                if [ "$SECONDS" -gt "$SSH_TIMEOUT" ]; then
+                    die "timeout accessing ${ip} through ssh"
```

Comment:
This is the same error as above, and it can be a bit misleading if (for whatever reason) ssh works, but the `/proc/cmdline` change didn't. Can you change it into `timeout waiting for ${ip} to reboot with new kernel parameters`?